.pre-commit-config.yaml

```diff
@@ -25,8 +25,8 @@ repos:
         files: ^(sqlglot/|tests/|setup.py)
       - id: mypy
         name: mypy
-        entry: mypy
+        entry: mypy sqlglot tests
```

Comment:
you were previously running `mypy` on every file that changed, but typing changes can't be isolated per file, hence mypy had to typecheck the entire code base of each file changed.

This should be much faster.